File: domain/src/main/java/com/hanghae/domain/entity/Movie.java
package com.hanghae.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Movie extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer movieId; // 영화 ID

    @Column(nullable = false, length = 100)
    private String title; // 영화 제목

    @Column(length = 50)
    private String rating; // 영상물 등급

    private LocalDate releaseDate; // 개봉일

    private String thumbnailUrl; // 썸네일 이미지(URL)

    private Integer duration; // 러닝 타임 (분)

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "genre_id", nullable = false)
    private Genre genre; // 장르 (외래키)

    @OneToMany(mappedBy = "movie", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Showtime> showtimes; // 상영 일정 (연관 관계)
}

File: domain/src/main/kotlin/io/github/jehwanyoo/redis_1st/model/Movie.kt
package io.github.jehwanyoo.redis_1st.model

import java.time.LocalDate

data class Movie(
    val id: Long,                   // 고유 ID
    val title: String,              // 영화 제목
    val releaseDate: LocalDate,     // 개봉일
    val thumbnailUrl: String,       // 썸네일 URL
    val runtimeMinutes: Int,        // 러닝 타임 (분)
    val genre: String,              // 장르
    val rating: String,             // 영상물 등급
)
